zeppelin-web/src/app/search/result-list.controller.js

```diff
@@ -34,6 +36,18 @@ angular
 
       return note;
     });
+    if ($scope.notes.length === 0) {
+      $scope.isResult = false;
+    } else {
+      $scope.isResult = true;
+    }
+
+    var routeChangeEvent = $rootScope.$on('$routeChangeStart', function (event, next, current) {
+      if (next.originalPath !== '/search/:searchTerm') {
+        angular.element('#searchTermId').val('');
+        routeChangeEvent();
```

Comment:
@felizbear by calling routeChangeEvent I am explicitly unbinding the event handler.
$routeScope.$on returns a deregistration function whose reference we are storing in routeChangeEvent.